Assets/Scripts/PlayerController/PlayerAnimation.cs

```diff
@@ -13,6 +13,11 @@ public class PlayerAnimation : MonoBehaviour
 
     private PlayerMovement m_PlayerMovement;
 
+    private const float m_doubleBuffDuration = 5.0f;
+
+    private bool m_hasJumpPower;
+    private bool m_hasSpeedPower;
+
```

Comment:
Consider creating a new script to handle powerups. Let `PlayerAnimation` deal with animations alone.
`PlayerAnimation` can then read state from the new script.

Reply:
Sures